src/components/Mailbox.vue

```diff
@@ -137,6 +137,35 @@ export default {
 		},
 	},
 	watch: {
+		// Force the envelope list to re-render when user clicks on the folder it's currently in
+		$route(to) {
+			if (to.params.threadId === undefined) {
+				const first = this.envelopes[0]
+				if (typeof first !== 'undefined') {
+					logger.debug('refreshing mailbox')
+					if (first.flags.draft) {
```

Comment:
I have mixed feelings about this. can we take the account's `draftsMailboxId` for a more reliable decision?

Reply:
yes, but in this case we should probably also change the test at line 210 (in master):

https://github.com/nextcloud/mail/blob/f3c5de02b387081270bdbc6103c2fe4472e7dc12/src/components/Mailbox.vue#L210